src/frontend/components/home/FeedCard.jsx

```diff
@@ -17,41 +20,96 @@ const FeedCard = ({ pInfo}) => {
         else {
             setLikedDisplay(() => false)
         }
+      
+
     },[pInfo])
-    const likeHandler = async() => {
-        if(!likedDisplay){
+    useEffect(() => {
+        const isPostBookmarked = bookmarksData.find((bookmark) => bookmark._id === pInfo._id)
+        if(isPostBookmarked){
+            setBookmarkDisplay(() => true)
+        }
+        else {
+            setBookmarkDisplay(() => false)
+        }
+    },[bookmarksData])
+    useEffect(() => {
+        const isFollowing = userProfileData.following.find((user) => user.firstName + user.lastName === postUploader.firstName + postUploader.lastName)
+        if(isFollowing){
+            setFollowDisplay(() =>true)
+        }
+        else {
+            setFollowDisplay(() =>false)
+        }
+
+    },[userProfileData])
+    const likeHandler = likeHandlerFunction(likedDisplay, pInfo, jwtToken, dispatch)
+    const bookmarkPost = async() => {
+        if(bookmarkDisplay){
             try {
-                const response = await axios.post(`/api/posts/like/${pInfo._id}`, {}, {headers:{authorization:jwtToken}})
-                dispatch({type:"SET_POSTS_DATA", payload:response.data.posts})
-    
-            }catch (e) {
+                const response = await axios.post(`/api/users/remove-bookmark/${pInfo._id}`,{}, { headers: { authorization: jwtToken } })
+                
+                dispatch({ type: "SET_BOOKMARK_DATA", payload: response.data.bookmarks });
+            }
+            catch (e) {
                 console.log(e)
             }
         }
         else {
             try {
-                const response = await axios.post(`/api/posts/dislike/${pInfo._id}`, {}, {headers:{authorization:jwtToken}})
-                dispatch({type:"SET_POSTS_DATA", payload:response.data.posts})
-    
-            }catch (e) {
+                const response = await axios.post(`/api/users/bookmark/${pInfo._id}`,{}, { headers: { authorization: jwtToken } })
+                console.log(response.data.bookmarks)
+                dispatch({ type: "SET_BOOKMARK_DATA", payload: response.data.bookmarks });
+            }
+            catch (e) {
+                console.log(e)
+            }
+        }
+       
+    }
+    const followHandler = async() => {
+        if(followDisplay){
```

Comment:
Instead of writing try and catch in conditions , you can take a variable url which can be set conditionally and rest of the call can be written once , hense reducing the code.